File: MiniDOMOffice/Office/Lavoro/Candidature/CandidaturaCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;


namespace minidom
{
    public partial class Office
    {

        /// <summary>
        /// Cursore di <see cref="Candidatura"/>
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class CandidaturaCursor 
            : minidom.Databases.DBObjectCursorPO<Candidatura>
        {
            private DBCursorField<DateTime> m_DataCandidatura = new DBCursorField<DateTime>("DataCandidatura");
            private DBCursorField<int> m_IDCandidato = new DBCursorField<int>("IDCandidato");
            private DBCursorStringField m_NomeCandidato = new DBCursorStringField("NomeCandidato");
            private DBCursorField<int> m_IDCurriculum = new DBCursorField<int>("IDCurriculum");
            private DBCursorField<int> m_IDOfferta = new DBCursorField<int>("IDOfferta");
            private DBCursorStringField m_NomeOfferta = new DBCursorStringField("NomeOfferta");
            private DBCursorField<int> m_IDCanale = new DBCursorField<int>("IDCanale");
            private DBCursorStringField m_NomeCanale = new DBCursorStringField("NomeCanale");
            private DBCursorStringField m_TipoFonte = new DBCursorStringField("TipoFonte");
            private DBCursorField<int> m_IDFonte = new DBCursorField<int>("IDFonte");
            private DBCursorStringField m_NomeFonte = new DBCursorStringField("NomeFonte");
            private DBCursorField<DateTime> m_DataNascita = new DBCursorField<DateTime>("DataNascita");
            private DBCursorStringField m_NatoA_Citta = new DBCursorStringField("NatoA_Comune");
            private DBCursorStringField m_NatoA_Provincia = new DBCursorStringField("NatoA_Provincia");
            private DBCursorStringField m_ResidenteA_Citta = new DBCursorStringField("ResidenteA_Citta");
            private DBCursorStringField m_ResidenteA_CAP = new DBCursorStringField("ResidenteA_CAP");
            private DBCursorStringField m_ResidenteA_Provincia = new DBCursorStringField("ResidenteA_Provincia");
            private DBCursorStringField m_ResidenteA_Via = new DBCursorStringField("ResidenteA_Via");
            private DBCursorStringField m_ResidenteA_Civico = new DBCursorStringField("ResidenteA_Civico");
            private DBCursorStringField m_Telefono = new DBCursorStringField("Telefono");
            private DBCursorStringField m_eMail = new DBCursorStringField("eMail");
            private DBCursorField<int> m_Valutazione = new DBCursorField<int>("Valutazione");
            private DBCursorField<int> m_ValutatoDaID = new DBCursorField<int>("ValutatoDaID");
            private DBCursorStringField m_ValutatoDaNome = new DBCursorStringField("ValutatoDaNome");
            private DBCursorField<DateTime> m_ValutatoIl = new DBCursorField<DateTime>("ValutatoIl");
            private DBCursorStringField m_MotivoValutazione = new DBCursorStringField("MotivoValutazione");

            /// <summary>
            /// Costruttore
            /// </summary>
            public CandidaturaCursor()
            {
            }

            /// <summary>
            /// DataNascita
            /// </summary>
            public DBCursorField<DateTime> DataNascita
            {
                get
                {
                    return m_DataNascita;
                }
            }

            /// <summary>
            /// NatoA_Citta
            /// </summary>
            public DBCursorStringField NatoA_Citta
            {
                get
                {
                    return m_NatoA_Citta;
                }
            }

            /// <summary>
            /// NatoA_Provincia
            /// </summary>
            public DBCursorStringField NatoA_Provincia
            {
                get
                {
                    return m_NatoA_Provincia;
                }
            }

            /// <summary>
            /// ResidenteA_Citta
            /// </summary>
            public DBCursorStringField ResidenteA_Citta
            {
                get
                {
                    return m_ResidenteA_Citta;
                }
            }

            /// <summary>
            /// ResidenteA_CAP
            /// </summary>
            public DBCursorStringField ResidenteA_CAP
            {
                get
                {
                    return m_ResidenteA_CAP;
                }
            }

            /// <summary>
            /// ResidenteA_Provincia
            /// </summary>
            public DBCursorStringField ResidenteA_Provincia
            {
                get
                {
                    return m_ResidenteA_Provincia;
                }
            }

            /// <summary>
            /// ResidenteA_Via
            /// </summary>
            public DBCursorStringField ResidenteA_Via
            {
                get
                {
                    return m_ResidenteA_Via;
                }
            }

            /// <summary>
            /// ResidenteA_Civico
            /// </summary>
            public DBCursorStringField ResidenteA_Civico
            {
                get
                {
                    return m_ResidenteA_Civico;
                }
            }

            /// <summary>
            /// Telefono
            /// </summary>
            public DBCursorStringField Telefono
            {
                get
                {
                    return m_Telefono;
                }
            }

            /// <summary>
            /// eMail
            /// </summary>
            public DBCursorStringField eMail
            {
                get
                {
                    return m_eMail;
                }
            }

            /// <summary>
            /// Valutazione
            /// </summary>
            public DBCursorField<int> Valutazione
            {
                get
                {
                    return m_Valutazione;
                }
            }

            /// <summary>
            /// ValutatoDaID
            /// </summary>
            public DBCursorField<int> ValutatoDaID
            {
                get
                {
                    return m_ValutatoDaID;
                }
            }

            /// <summary>
            /// ValutatoDaNome
            /// </summary>
            public DBCursorStringField ValutatoDaNome
            {
                get
                {
                    return m_ValutatoDaNome;
                }
            }

            /// <summary>
            /// ValutatoIl
            /// </summary>
            public DBCursorField<DateTime> ValutatoIl
            {
                get
                {
                    return m_ValutatoIl;
                }
            }

            /// <summary>
            /// MotivoValutazione
            /// </summary>
            public DBCursorStringField MotivoValutazione
            {
                get
                {
                    return m_MotivoValutazione;
                }
            }

            /// <summary>
            /// IDCanale
            /// </summary>
            public DBCursorField<int> IDCanale
            {
                get
                {
                    return m_IDCanale;
                }
            }

            /// <summary>
            /// NomeCanale
            /// </summary>
            public DBCursorStringField NomeCanale
            {
                get
                {
                    return m_NomeCanale;
                }
            }

            /// <summary>
            /// TipoFonte
            /// </summary>
            public DBCursorStringField TipoFonte
            {
                get
                {
                    return m_TipoFonte;
                }
            }

            /// <summary>
            /// IDFonte
            /// </summary>
            public DBCursorField<int> IDFonte
            {
                get
                {
                    return m_IDFonte;
                }
            }

            /// <summary>
            /// NomeFonte
            /// </summary>
            public DBCursorStringField NomeFonte
            {
                get
                {
                    return m_NomeFonte;
                }
            }

            /// <summary>
            /// DataCandidatura
            /// </summary>
            public DBCursorField<DateTime> DataCandidatura
            {
                get
                {
                    return m_DataCandidatura;
                }
            }

            /// <summary>
            /// IDCandidato
            /// </summary>
            public DBCursorField<int> IDCandidato
            {
                get
                {
                    return m_IDCandidato;
                }
            }

            /// <summary>
            /// NomeCandidato
            /// </summary>
            public DBCursorStringField NomeCandidato
            {
                get
                {
                    return m_NomeCandidato;
                }
            }

            /// <summary>
            /// IDOfferta
            /// </summary>
            public DBCursorField<int> IDOfferta
            {
                get
                {
                    return m_IDOfferta;
                }
            }

            /// <summary>
            /// NomeOfferta
            /// </summary>
            public DBCursorStringField NomeOfferta
            {
                get
                {
                    return m_NomeOfferta;
                }
            }

            /// <summary>
            /// IDCurriculum
            /// </summary>
            public DBCursorField<int> IDCurriculum
            {
                get
                {
                    return m_IDCurriculum;
                }
            }

            /// <summary>
            /// Restituisce l'elenco dei campi da inserire nel cursore
            /// </summary>
            /// <returns></returns>
            protected override DBCursorFieldsCollection GetCursorFields()
            {
                var ret = base.GetCursorFields();
                ret.Remove(this.m_Telefono);
                return ret;
            }
 
            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Office.Candidature;
            }

            /// <summary>
            /// Restituisce la clausola where
            /// </summary>
            /// <returns></returns>
            public override DBCursorFieldBase GetWherePart()
            {
                var ret = base.GetWherePart();
                if (m_Telefono.IsSet())
                {
                    //TODO usare il campo intero (specializzare il campo con il nuovo nome)
                    ret *= (this.Field("Telefono1").EQ(this.m_Telefono.Value))
                           + (this.Field("Telefono2").EQ(this.m_Telefono.Value))
                            ;
                }

                return ret;
            }
        }
    }
}

File: MiniDOMOffice/Office/Lavoro/Candidature/CandidaturaCursor.vb
Imports minidom.Anagrafica
Imports minidom.Databases
Imports minidom.Sistema


Partial Class Office

    ''' <summary>
    ''' Cursore sulla tabella delle candidature ad un'offerta di lavoro
    ''' </summary>
    ''' <remarks></remarks>
    Public Class CandidaturaCursor
        Inherits DBObjectCursorPO(Of Candidatura)

        Private m_DataCandidatura As New CCursorField(Of Date)("DataCandidatura")
        Private m_IDCandidato As New CCursorField(Of Integer)("IDCandidato")
        Private m_NomeCandidato As New CCursorFieldObj(Of String)("NomeCandidato")
        Private m_IDCurriculum As New CCursorField(Of Integer)("IDCurriculum")
        Private m_IDOfferta As New CCursorField(Of Integer)("IDOfferta")
        Private m_NomeOfferta As New CCursorFieldObj(Of String)("NomeOfferta")
        Private m_IDCanale As New CCursorField(Of Integer)("IDCanale")
        Private m_NomeCanale As New CCursorFieldObj(Of String)("NomeCanale")
        Private m_TipoFonte As New CCursorFieldObj(Of String)("TipoFonte")
        Private m_IDFonte As New CCursorField(Of Integer)("IDFonte")
        Private m_NomeFonte As New CCursorFieldObj(Of String)("NomeFonte")
        Private m_DataNascita As New CCursorField(Of Date)("DataNascita")
        Private m_NatoA_Citta As New CCursorFieldObj(Of String)("NatoA_Comune")
        Private m_NatoA_Provincia As New CCursorFieldObj(Of String)("NatoA_Provincia")
        Private m_ResidenteA_Citta As New CCursorFieldObj(Of String)("ResidenteA_Citta")
        Private m_ResidenteA_CAP As New CCursorFieldObj(Of String)("ResidenteA_CAP")
        Private m_ResidenteA_Provincia As New CCursorFieldObj(Of String)("ResidenteA_Provincia")
        Private m_ResidenteA_Via As New CCursorFieldObj(Of String)("ResidenteA_Via")
        Private m_ResidenteA_Civico As New CCursorFieldObj(Of String)("ResidenteA_Civico")
        Private m_Telefono As New CCursorFieldObj(Of String)("Telefono")
        Private m_eMail As New CCursorFieldObj(Of String)("eMail")
        Private m_Valutazione As New CCursorField(Of Integer)("Valutazione")
        Private m_ValutatoDaID As New CCursorField(Of Integer)("ValutatoDaID")
        Private m_ValutatoDaNome As New CCursorFieldObj(Of String)("ValutatoDaNome")
        Private m_ValutatoIl As New CCursorField(Of Date)("ValutatoIl")
        Private m_MotivoValutazione As New CCursorFieldObj(Of String)("MotivoValutazione")

        Public Sub New()
        End Sub

        Public ReadOnly Property DataNascita As CCursorField(Of Date)
            Get
                Return Me.m_DataNascita
            End Get
        End Property

        Public ReadOnly Property NatoA_Citta As CCursorFieldObj(Of String)
            Get
                Return Me.m_NatoA_Citta
            End Get
        End Property

        Public ReadOnly Property NatoA_Provincia As CCursorFieldObj(Of String)
            Get
                Return Me.m_NatoA_Provincia
            End Get
        End Property

        Public ReadOnly Property ResidenteA_Citta As CCursorFieldObj(Of String)
            Get
                Return Me.m_ResidenteA_Citta
            End Get
        End Property

        Public ReadOnly Property ResidenteA_CAP As CCursorFieldObj(Of String)
            Get
                Return Me.m_ResidenteA_CAP
            End Get
        End Property

        Public ReadOnly Property ResidenteA_Provincia As CCursorFieldObj(Of String)
            Get
                Return Me.m_ResidenteA_Provincia
            End Get
        End Property

        Public ReadOnly Property ResidenteA_Via As CCursorFieldObj(Of String)
            Get
                Return Me.m_ResidenteA_Via
            End Get
        End Property

        Public ReadOnly Property ResidenteA_Civico As CCursorFieldObj(Of String)
            Get
                Return Me.m_ResidenteA_Civico
            End Get
        End Property

        Public ReadOnly Property Telefono As CCursorFieldObj(Of String)
            Get
                Return Me.m_Telefono
            End Get
        End Property

        Public ReadOnly Property eMail As CCursorFieldObj(Of String)
            Get
                Return Me.m_eMail
            End Get
        End Property

        Public ReadOnly Property Valutazione As CCursorField(Of Integer)
            Get
                Return Me.m_Valutazione
            End Get
        End Property

        Public ReadOnly Property ValutatoDaID As CCursorField(Of Integer)
            Get
                Return Me.m_ValutatoDaID
            End Get
        End Property

        Public ReadOnly Property ValutatoDaNome As CCursorFieldObj(Of String)
            Get
                Return Me.m_ValutatoDaNome
            End Get
        End Property

        Public ReadOnly Property ValutatoIl As CCursorField(Of Date)
            Get
                Return Me.m_ValutatoIl
            End Get
        End Property

        Public ReadOnly Property MotivoValutazione As CCursorFieldObj(Of String)
            Get
                Return Me.m_MotivoValutazione
            End Get
        End Property

        Public ReadOnly Property IDCanale As CCursorField(Of Integer)
            Get
                Return Me.m_IDCanale
            End Get
        End Property

        Public ReadOnly Property NomeCanale As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeCanale
            End Get
        End Property

        Public ReadOnly Property TipoFonte As CCursorFieldObj(Of String)
            Get
                Return Me.m_TipoFonte
            End Get
        End Property

        Public ReadOnly Property IDFonte As CCursorField(Of Integer)
            Get
                Return Me.m_IDFonte
            End Get
        End Property

        Public ReadOnly Property NomeFonte As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeFonte
            End Get
        End Property

        Public ReadOnly Property DataCandidatura As CCursorField(Of Date)
            Get
                Return Me.m_DataCandidatura
            End Get
        End Property

        Public ReadOnly Property IDCandidato As CCursorField(Of Integer)
            Get
                Return Me.m_IDCandidato
            End Get
        End Property

        Public ReadOnly Property NomeCandidato As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeCandidato
            End Get
        End Property

        Public ReadOnly Property IDOfferta As CCursorField(Of Integer)
            Get
                Return Me.m_IDOfferta
            End Get
        End Property

        Public ReadOnly Property NomeOfferta As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeOfferta
            End Get
        End Property

        Public ReadOnly Property IDCurriculum As CCursorField(Of Integer)
            Get
                Return Me.m_IDCurriculum
            End Get
        End Property

        Protected Overrides Function GetWhereFields() As CKeyCollection(Of CCursorField)
            Dim ret As CKeyCollection(Of CCursorField) = MyBase.GetWhereFields()
            ret.RemoveByKey("Telefono")
            Return ret
        End Function

        Protected Overrides Function GetConnection() As CDBConnection
            Return Office.Database
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Office.Candidature.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_OfficeCandidature"
        End Function

        Public Overrides Function GetWherePart() As String
            Dim ret As String = MyBase.GetWherePart()
            If (Me.m_Telefono.IsSet) Then
                Dim fSQL As String = Me.m_Telefono.GetSQL
                Dim f1 As String = Replace(fSQL, "[Telefono]", "[Telefono1]")
                Dim f2 As String = Replace(fSQL, "[Telefono]", "[Telefono2]")
                ret = Strings.Combine(ret, "(" & f1 & " OR " & f2 & ")", " AND ")
            End If
            Return ret
        End Function


    End Class


End Class